docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,226 @@
+# Formatted ORAS CLI output 
+
+ORAS has prettified output designed for humans. However, for machine processing, especially in some automation scenarios like scripting and CI/CD pipelines, developers want to perform batch operations and chain different commands with ORAS, as well as filtering, modifying, and sorting objects based on the outputs that are emitted by the ORAS command. Developers expect the ORAS output can be generated as machine-readable text not only the prettified or tabular data, so that they can use the formatted outputs like JSON output to perform further advanced operations. 
+
+The formatted output is not intended to supersede the prettified human-readable and friendly output text of ORAS CLI. It aims to provide programming-friendly experience for developers who want to automate their workflows on machines especially on Unix and most Unix-like operating systems, without parsing unstructured text. It will increase the developer experience for ORAS in automation and scripting scenarios. 
+
+## Scenarios
+
+### Scripting
+
+Alice is a developer who wants to batch operations with ORAS in her Shell script. In order to automate some routine workflow in containers secure supply chain scenario, she wants the machine to get the image digest from the JSON output objects that are emitted by `oras push`, then use utility like [xargs](https://en.wikipedia.org/wiki/Xargs) to enable an ORAS command to act on the output of another command and perform further steps. In this way, she can chain commands together. For example, she can use `oras attach` to attach an SBOM to the image using its image digest as a argument outputted from `oras push`.
+
+For example, push an artifact to a registry and generate the artifact reference in the standard output, then attach an SBOM to the artifact using the artifact reference (`$REG/$REPO:$DIGEST`) outputted from the first command, finally sign the attached SBOM with another tool against the SBOM file's reference (`$REG/$REPO:$DIGEST`) outputted from the last step.
+
+```shell
+oras push $REG/$REPO:$TAG hello.txt --format '{{.Reference}}' |\
+xargs -I _ oras attach --artifact-type sbom/example _ sbom.spdx --format '{{.Reference}}' |\
+xargs -I _ notation sign _ 
+```
```

Comment:
Extended arguments utility(xarg) is only available on *nix platforms. We also need to add PowerShell commands (like `ConvertTo-Json`）for Windows platform?

Reply:
I will add PowerShell command example for Windows platform

Reply:
I would recommend minimizing being dependent on any external tools, like `xargs` and instead provide as much tooling integrated with the tool

Reply:
Agree with @sabre1041 . I will add another sample commands for using environmental variables here.

Reply:
Added another two samples for using environmental variables and [ConvertFrom-Json](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-json)  Windows PowerShell in this scenario,